geoportailv3/mymaps.py

```diff
@@ -0,0 +1,321 @@
+                    ob = geom.__geo_interface__
+                else:
+                    ob = geom
+                geom_type = ob.get("type").lower()
+
+                if types is None:
+                    types = geom_type
+                else:
+                    if types == geom_type:
+                        is_transformable = True
+                    else:
+                        is_transformable = False
+                        break
+                geoms.append(asShape(geom))
+            if is_transformable:
+                if types == "point":
+                    shape = asMultiPoint(geoms)
+                elif types == "linestring":
+                    shape = asMultiLineString(geoms)
+                elif types == "polygon":
+                    shape = MultiPolygonAdapter(geoms, context_type='geojson')
+            else:
+                shape = None
+
+        self.geometry = from_shape(shape, srid=2169)
+
+    @property
+    def __geo_interface__(self):
+        geometry = wkb.loads(str(self.geometry), True)
+        properties = dict(name=self.name,
+                          description=self.description,
+                          thumbnail=self.thumbnail,
+                          image=self.image,
+                          color=self.color,
+                          stroke=self.stroke,
+                          isLabel=self.is_label,
+                          showOrientation=self.show_orientation,
+                          linestyle=self.linestyle,
+                          id=self.id,
+                          symbol_id=self.symbol_id,
+                          angle=self.angle,
+                          size=self.size,
+                          font_size=self.font_size
+                          )
+        return geojson.Feature(id=self.id,
+                               geometry=geometry,
+                               properties=properties
+                               )
+
+    @property
+    def geom(self):
+        if hasattr(self.geometry, "geom_wkb"):
+            return wkb.loads(str(self.geometry.geom_wkb))
+        else:
+            if hasattr(self, "_shape"):
+                return self._shape
+            else:
+                return None
+
+role_categories = Table('role_categories',
+                        Base.metadata,
+                        Column('role_id',
+                               Integer,
+                               ForeignKey('role.id')),
+                        Column('category_id',
+                               Integer,
+                               ForeignKey('category.id')))
+
+
+class Category(Base):
+    __tablename__ = 'category'
+
+    id = Column(Integer, primary_key=True)
+    name = Column(Unicode(255))
+    allow_labeling = Column(Boolean)
+
+    def __init__(self, name):
+        self.name = name
+
+    def __allow_labeling__(self):
+        if (self.allow_labeling):
+            return True
+        else:
+            return False
```

Comment:
Ad what's the point of that function? Why just not use `self.allow_labeling` in `todict`?